swap_meet/vendor.py

```diff
@@ -1,2 +1,69 @@
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not other_vendor.inventory or not self.inventory:
+            return False
+        
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        first_item = self.inventory[0]
+        other_vendor_first_item = other_vendor.inventory[0]
+
+        self.swap_items(other_vendor, first_item, other_vendor_first_item)        
+        return True
+
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+        return category_list
```

Comment:
This would be a great opportunity for a list comprehension. This is equivalent to

```python
category_list = [item for item in self.inventory if item.category == category]